llvm/lib/SYCL/PrepareSYCLOpt.cpp

```diff
@@ -171,13 +173,50 @@ struct PrepareSYCLOpt : public ModulePass {
     }
   }
 
+  struct UnwrapperVisitor : public llvm::InstVisitor<UnwrapperVisitor> {
+    void visitCallInst(CallInst &I) {
+      auto *ParentF = I.getFunction();
+      auto *F = I.getCalledFunction();
+      if (!F->hasFnAttribute("fpga.propertywrapper"))
+        return;
+      // We have a property wrapper.
```

Comment:
What is a property wrapper should be explained somewhere describing the design

Reply:
I've added some comment here, but I don't know where the "top level doc" should go.